File: eval/2020_07_02/part3_fruits_structure.cs
/*
Résultat attendu
*/
class Fruit
{
    private string label;

    private float price;

    public string GetLabel() // With C#, all methods use PascalCase
    {

    }

    public float GetPrice()
    {

    }

    public void SetPrice(float price)
    {

    }
}


class Pear : Fruit
{
    private string color;

    public string GetColor()
    {

    }

    public void SetColor(string color)
    {

    }
}

class Apple : Fruit 
{
    private bool redlabel;

    public bool IsRedLabel()
    {

    }
}

class Customer
{
    private int clientId;

    private Fruit sample;

    public Customer(int clientId)
    {

    }

    public int GetClientId()
    {

    }
}

class Basket
{
    private int @ref; // ref = C# reserved keyword. "@" prefix allows to use a keyword as class attribute name

    private Fruit[] items;

    public Basket(int _ref, Fruit fruit)
    {

    }

    public float GetTotalPrice()
    {
        
    }

}

File: projets/evaluations/2020_07_02/part3_fruits_structure.js
/*
Résultat attendu
*/



class Fruit
{
    /** @var string label */
    #label;

    /** @var Number price */
    #price;

    getLabel() 
    {

    }

    getPrice()
    {

    }

    setPrice(price)
    {

    }
}



class Pear extends Fruit
{
    /** @var string color */
    #color;

    getColor()
    {

    }

    setColor(color)
    {

    }
}



class Apple extends Fruit 
{
    /** @var bool redLabel */
    #redlabel;

    isRedLabel()
    {

    }
}



class Customer
{
    /** @var Number clientId */
    #clientId;

    /** @var Fruit sample */
    #sample;

    constructor(clientId)
    {

    }

    getClientId()
    {

    }
}



class Basket
{
    /** @var Number ref */
    #ref; 

    /** @var Fruit[] items */
    #items;

    constructor(ref, fruit)
    {

    }

    getTotalPrice()
    {
        
    }
}